src/eth/rpc/rpc_subscriptions.rs

```diff
@@ -33,16 +33,14 @@ pub struct RpcSubscriptions {
 impl RpcSubscriptions {
     /// Spawns a new thread to clean up closed subscriptions from time to time.
     pub fn spawn_subscriptions_cleaner(self: Arc<Self>) {
-        tokio::spawn(async move {
-            loop {
-                for closed_sub in self.new_heads.iter().filter(|sub| sub.is_closed()) {
-                    self.new_heads.remove(closed_sub.key());
-                }
-                for closed_sub in self.logs.iter().filter(|sub| sub.0.is_closed()) {
-                    self.logs.remove(closed_sub.key());
-                }
-                sleep(CLEANING_FREQUENCY).await;
+        tokio::task::spawn_blocking(move || loop {
```

Comment:
Use spawn_blocking instead of spawn because Dashmap was causing deadlocks when running in the async context.